### DISCUSSION TITLE
Inconsistent redirects

### DISCUSSION THREAD

**Discussion by alexklibisz** [firefly-iii/firefly-iii#4834] [Firefly III]
I am running Firefly III version 5.5.11.

**Description**

I find firefly-iii is often inconsistent with the pages that it redirects you to when completing some action.

Example: I'm creating a new rule. I start from the rules page and click "new rule". I create the rule, and it takes me back to the rules page. This makes sense. Then I start creating another rule. Exact same process. I create the rule, and it takes me to the home page, or a recent transactions view for a specific account. These are generally pages that I've recently viewed, but there doesn't seem to be any obvious pattern.

**Extra info**

- I'm using the older UI.
- I'm running FF3 in a docker container with an nginx reverse proxy in front of the container.

Perhaps this is an issue with my reverse proxy configuration? But I also self-host several other apps behind identical proxies, and haven't seen the issue there. Maybe it's also a known/solved issue. I searched around but perhaps I'm not describing it in a very standard way.

**Comment by JC5** [comment]
It will be partially fixed in the new layout, but until the app is a SPA it's not something I can do much about.

Firefly III remembers the page you came from. It remembers that page whenever you're on a form, so it can redirect you back.

- If your browser decides not to submit referrals, you will always go back to the index because Firefly III doesn't know where you came from.
- If you have multiple tabs open, Firefly III may remember the wrong page.
- If you would have been directed to a page that's a 404, you may be sent to the index instead.

The code itself is pretty solid and I know it works reliabily. However, between browser quircks and security settings (also headers like `Referrer-Policy`) it may fail occasionally.

**Reply by spartan117aut** [reply]
> If you have multiple tabs open, Firefly III may remember the wrong page.

This. This behaviour confuses me regularly when using multiple tabs. I hope it will be included in the things that will be fixed in the new layout. 
Thank you for your work!

**Reply by alexklibisz** [reply]
@spartan117aut Are you also using nginx or some other reverse proxy?

**Reply by spartan117aut** [reply]
@alexklibisz No, just a docker container using the provided docker-compose.yml.

**Comment by alexklibisz** [comment]
Is there any update on this? It's still broken on the latest version (5.7.15), and it seems the V2 layout never quite landed?
